0_functions/simulation_functions.R

```diff
@@ -261,3 +261,78 @@ betabinom_cdf <- function(x, size, p, phi) rmutil::pbetabinom(x, size, p, phi)
 binom_pmf <- function(x, size, p) dbinom(x, size, p)
 betabinom_pmf <- function(x, size, p, phi) rmutil::dbetabinom(x, size, p, phi)
 
+
+
+# load and modify the BBC pandemic data as we need to for the draw_gatherings() function below
+bbc <- read_csv("1_data/contact_dist_BBCPandemic/contact_distributions_o18.csv")
+bbc <- select(bbc, e_other)
+# gathering sizes are contacts + 1 (index person)
+bbc$M <- bbc$e_other + 1
+# plotting it
+bbc_data <- ggplot(data = bbc, mapping = aes(x = M)) + 
+  geom_point(stat = "count") + 
+  scale_y_log10() + scale_x_log10()
+pdf("3_results/bbc_data.pdf", width = 5, height = 6)
+print(bbc_data)
+dev.off()  
+
+# draw_gatherings() function generates M, a vector of gatherings drawn from the
+# BBC pandemic distribution so that in total N people attend,
+# under 4 options for restriction in gathering size up to c : 0 for no restriction, 
+# 1 for replacement by c, 2 for replacement by 0 and 3 for replacement by another draw smaller to c
+
+draw_gatherings <- function(N = 10000, #10000 as default, can change
+                            c = 10, # cut-off as default, can change
+                            option) { # define options for restriction (1, 2, or 3)
+# initialize loop variables
+M <- vector()
+i <- 1
+# create subset of bbc distribution <= c (for option 3)
+smallM <- sample(bbc$M[bbc$M <= c])
+# loop until sum of gatherings is equal to pop size
+while(sum(M) != N) {
+  # draw gathering of size m from BBC pandemic distribution
+  M[i] <- sample(bbc$M, 1) 
+  if (M[i] > c) {
+    M[i] <- switch(
+      option,
+      # option 0 : no restriction to gathering size 
+      "0" = M[i],
+      # option 1: set to c (most conservative)
+      "1" = c,
+      # option 2: set to 1 (least conservative)
+      "2" = 1,
+      # option 3: redraw until less than c (realistic?) 
+      #"3" = while (M[i] > c) # Chris I'm not sure what you were trying to do here (and it didn't seem to work?)
+      #  sample(bbc$M, 1)
+      "3" = sample(smallM, 1) #So I did that instead
```

Comment:
This works better than what I had, I like it... although no need to first define `smallM` and then do this line. just make this line 
```
"3" = sample(bbc$M[bbc$M <=c], 1)
```